stock_reorder_forecast/models/purchase_order.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    # TODO UPDATE PURCHASE PROPOSAL UPDATE RFQ
+    @api.multi
+    def update_proposal(self, upd_ids=None):
+        # if there is a draft PO with this product  i
+        # set ultimate_purchase ==NULL.
+        if not upd_ids:
+            upd_ids = []
+            for this in self:
+                upd_ids.append(this.id)
+        # removing queries when not strictly needed, using ORM
+        rfqs = self.env['purchase.order'].search([
+            ('id', 'in', upd_ids),
+            ('state', '=', 'draft')
+        ])
+        for rfq in rfqs:
+            for pol in rfq.order_line:
+                pol.product_id.write({'ultimate_purchase': False})
+        # set the partner's ultimate_purchase as the soonest
+        # of all the UP's in the products where he is supplier.
+        sql = self.env['res.partner'].sql_update_partner
```

Comment:
why isn't this a function that does whatever you need to be done on the partner?